app/api/server/api.js

```diff
@@ -588,9 +588,20 @@ const createApi = function _createApi(enableCors) {
 			defaultOptionsEndpoint,
 			auth: getUserAuth(),
 		});
+	} else {
+		API.v1._config.enableCors = enableCors;
+		if (enableCors) {
```

Comment:
to reduce duplicated code I'd recommend creating a function to do what [this block of code](https://github.com/RocketChat/Rocket.Chat/pull/15443/files#diff-e8a96717dbd3869af6e51ee232c94025R593-R601) does.